crates/aws/src/lib.rs

```diff
@@ -46,7 +46,18 @@ impl LogStoreFactory for S3LogStoreFactory {
         location: &Url,
         options: &StorageOptions,
     ) -> DeltaResult<Arc<dyn LogStore>> {
-        let store = url_prefix_handler(store, Path::parse(location.path())?)?;
+        let path_decoded = match urlencoding::decode(location.path()) {
```

Comment:
This line would benefit from a comment as to what this decoding does/why it exists

Reply:
Added